python/plugins/processing/algs/help/qgis.yaml

```diff
@@ -466,6 +466,8 @@ qgis:refactorfields: >
 
   The original layer is not modified. A new layer is generated, which contains a modified attribute table, according to the provided fields mapping.
 
+  Rows in orange are not allowed to be NULL in the layer used as target. However, since the algorithm creates a new layer with no constraints, this can be taken only as a hint.
```

Comment:
I really don't see the value of this given the listed limitation. It's very confusing for users, who would expect this flag to apply to the newly created layer.

Reply:
@nyalldawson, if you use the Refactor Fields algorithm, you know it's a bit overwhelming, this PR aims to make it easier for users to prioritize field mappings. 

We actually use the refactor fields along with an "Append Features to Layer" algorithm (3rd party plugin that I also suggested to port to the QGIS core) to send the result into an existing layer. So, it acts as an entire ETL algorithm where knowing the target field constraints makes (a lot of) sense, especially for layers with a large number of fields.

Reply:
I still don't see the value sorry, and foresee the inevitable bug reports incoming about how the value isn't reflected in the output layer. Maybe for your particular use case you are best off subclassing the existing algorithm into your plugin so that it can be designed to meet your particular use case, rather then changing the core one itself.

Reply:
I agree with @nyalldawson that this is very confusing.
This only give the information that the layer used as a template for fields have not null constraints, but the purpose of this algorithm is to create a new layer.
This would make sense in an algorithm that writes features in an existing layer.